### PR TITLE
feat: Add --json-extended option to mise env

### PR DESCRIPTION
**Pull request by hverlin**
Add `mise env --json-extended` which shows where each environment variable is sourced from.

see #3012

### CODE REVIEW

File: src/cli/env.rs [hunk 1]
```diff
@@ -51,6 +58,61 @@ impl Env {
...
     }
 
+    fn output_extended_json(&self, config: &Config, ts: Toolset) -> Result<()> {
+        let mut res = BTreeMap::new();
+
+        ts.env_with_path(config)?.iter().for_each(|(k, v)| {
+            res.insert(k.to_string(), BTreeMap::from([("value", v.to_string())]));
+        });
+
+        config.env_with_sources()?.iter().for_each(|(k, v)| {
```
Comment (hverlin) [inline src/cli/env.rs:68]: This is iterating twice as `env_with_path` already does the same. If this is an issue, we would need to move the logic to add source/tool into `env_with_path` which I was not sure about given that this is used everywhere

Comment (jdx) [reply src/cli/env.rs:68]: these things almost never make any practical difference in mise

**Comment by jdx** [comment]
your test is failing. I would pipe it to jq and just parse some of it

File: src/cli/env.rs (hunk 1 above)
Comment (hverlin) [reply src/cli/env.rs:68]: Is it ok to keep as is, or should I move the logic in `env_with_path`?

Comment (jdx) [reply src/cli/env.rs:68]: it's fine like this

**Pull request merged**